File: music-dlx-web/src/models/track.ts
import { IAlbum, IAlbumResult } from "./album";
import { IArtist, IArtistResult } from "./artist";
import { ILyrics } from "./lyrics";

export interface ITrack {
  id: number;
  title: string;
  album: IAlbum;
  artist: IArtist;
  contributors: IArtist[];
  duration: number;
  disk_number: number;
  track_number: number;
  release_date: string;
  lyrics?: ILyrics;
}

export interface ITrackResult {
  id: number;
  title: string;
  album: IAlbumResult;
  artist: IArtistResult;
  duration: number;
}

export type IAnyTrack = ITrack | ITrackResult;

File: server/models/track.py
from dataclasses import dataclass

from .album import Album, AlbumResult
from .artist import Artist, ArtistResult
from .lyrics import Lyrics


@dataclass
class Track:
    id: int
    title: str
    album: Album
    artist: Artist
    contributors: list[Artist]
    duration: int
    disk_number: int
    track_number: int
    release_date: str
    lyrics: Lyrics | None


@dataclass
class TrackResult:
    id: int
    title: str
    album: AlbumResult
    artist: ArtistResult
    duration: int
